Dockerfile

```diff
@@ -0,0 +1,43 @@
+# Build the image in a distinct stage so we don't need the Golang SDK.
+FROM golang:1.11-stretch as builder
+
+# Install pre-reqs.
+#     - Update apt-get sources
+RUN apt-get update -y
+#     - Dep, for Go package management
+RUN curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
```

Comment:
Note that we pick a fixed version of dep - and in fact not the latest major release  - in all of our existing builds. We likely want to match that here. (And update all of them to the latest stable build soon).

Reply:
I can certainly do this. It's not clear to me how to easily share logic for this, so if you have any suggestions, let me know, otherwise I'll just hard-code `0.4.1` in here.

I'm not sure we should be doing this in general, by the way, since it means we'll just update them very infrequently. Case in point, it appears we haven't touched them in months. This is unfortunate, as we're not automatically getting improvements (bug fixes, performance, etc).

Reply:
Actually was able to do a minor refactoring to share the versions between the Dockerfile and CI scripts. Not too bad. I wonder if we should schedule a recurring sprintly "chore" to look at these and bump to the latest where appropriate.